workflow/controller/workflowpod.go

```diff
@@ -266,11 +266,20 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 			return nil, errors.Wrap(err, "", "Error in Unmarshalling after merge the patch")
 		}
 	}
-	created, err := woc.controller.kubeclientset.CoreV1().Pods(woc.wf.ObjectMeta.Namespace).Create(pod)
+
+	created, err := woc.controller.kubeclientset.CoreV1().Pods(woc.wf.ObjectMeta.Namespace).Get(nodeID, metav1.GetOptions{})
```

Comment:
Instead of making API call to get Pods. Workflow object has `id` field in nodes under nodestatus. we can check the given `NodeID` already there or not, to make Pod create call which will reduce the API calls.

Reply:
where's the best place to add the check if the node id exists in the map yet? I'm seeing that the nodeId will already be in the map before we make a Create request the 1st time.

Reply:
adding an else case here works https://github.com/argoproj/argo/blob/master/workflow/controller/operator.go#L1255

```
+       } else {
+               return node, nil
```

would that be ok?

Reply:
I am also thinking the same but code has already checks. I am looking at which scenario this issue happening.
https://github.com/argoproj/argo/blob/723b3c15e55d2f8dceb86f1ac0a6dc7d1a58f10b/workflow/controller/operator.go#L1413

Reply:
@Syps I think we are trying to fix the side effect, not a root cause. I am looking at this issue. Which recent change is causing frequent WF/POD update. I will let you know the issue.

Reply:
much appreciated! Look forward to any updates.

Reply:
Close the PR. @simster7 created PR to refactor the changes. #1818